clippy_lints/src/attrs.rs

```diff
@@ -619,10 +667,16 @@ impl EarlyLintPass for EarlyAttributes {
     extract_msrv_attr!(EarlyContext);
 }
 
-fn check_empty_line_after_outer_attr(cx: &EarlyContext<'_>, item: &rustc_ast::Item) {
+/// Selectively checks for empty lines after outer attributes.
+///
+/// Attributes and documenation comments are both considered outer attributes
+/// by the AST. However, the average user likely considers them to be different.
+/// Checking for empty lines after each of these attributes is split into two different
+/// lints but can share the same logic.
+fn check_empty_line_after_outer_attr_kind(cx: &EarlyContext<'_>, item: &rustc_ast::Item, lint_doc_comment: bool) {
```

Comment:
There's no reason to run over the attributes twice here. Just lint all attributes on the same pass through. If the lint isn't enabled then the generated message is suppressed in the `span_lint` call.

Reply:
Ah cool, changes made!